### PR TITLE
GH-1310: Implemented auto-reveal feature in the navigator.

### LINKED ISSUE #1310: [navigator] Sync with editor
A 'Reveal in Files' command that is added to the context menu of the editor tabs, should reveal the file of the editor in the navigator widget.
A preference `navigator.autoReveal` shall be used to automatically reveal the element in the files navigator when activating an editor. It shall default to `true`.

### PR DESCRIPTION
**Pull request by kittaakos**
Signed-off-by: Akos Kitta <kittaakos@gmail.com>

Closes #1310

### CODE REVIEW

File: packages/navigator/src/browser/navigator-contribution.ts [hunk 1]
```diff
@@ -5,13 +5,50 @@
...
+import { ISelectableTreeNode } from '@theia/core/lib/browser';
+import { MenuModelRegistry } from '@theia/core/lib/common/menu';
+import { CommandRegistry } from '@theia/core/lib/common/command';
+import { ApplicationShell } from '@theia/core/lib/browser/shell/application-shell';
+import { SHELL_TABBAR_CONTEXT_MENU } from '@theia/core/lib/browser/shell/tab-bars';
+import { NavigatorPreferences } from './navigator-preferences';
+
+export namespace FileNavigator {
+
+    export type Revealable = { readonly uri: URI } | URI;
```
Comment (akosyakov) [inline packages/navigator/src/browser/navigator-contribution.ts]: to discuss: Could we use `SelectionService` and `UriSelection` instead?

File: packages/editor/src/browser/editor-widget.ts [hunk 2]
```diff
@@ -28,6 +29,10 @@ export class EditorWidget extends BaseWidget implements SaveableSource {
         return this.editor.document;
     }
 
+    get uri(): URI {
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply packages/editor/src/browser/editor-widget.ts]: The alternative is a strong dependency to navigator.

File: packages/navigator/src/browser/navigator-contribution.ts (hunk 1 above)
Comment (svenefftinge) [reply packages/navigator/src/browser/navigator-contribution.ts]: I don't think it is the same. What we are looking for is a widget that has been opened for a file and is active.

**Comment by kittaakos** [comment]
Closing in favor of #1375. (Let's keep the branch for a while.)

**Pull request closed without merging**